QPong-Unity-Version/Assets/Scripts/CircuitGridControl.cs

```diff
@@ -160,15 +160,15 @@ void Update()
                 {
                     int gate_index = i * columnMax + j;
                     if (gateArray[gate_index] == "I"){
-                        gateObjectArray[gate_index].GetComponent<SpriteRenderer>().sprite = emptyGateSprite;
+                        gateObjectArray[gate_index].GetComponent<Gate>().SetGateIcon(emptyGateSprite);
```

Comment:
This is very neat! I really had no idea how to call a method in Unity. Now I learn it. :)